File: DMOJ/CCC '00 J1 - Calendar.py
import math

data = input("")
info = data.split(" ")

weeks = math.ceil((int(info[1]) + int(info[0]) - 1) / 7)

first = True
current = 1
days = [[0]*7 for n in range(weeks)]

for x in range(0, weeks):
    for z in range(1, 8):
        if current > int(info[1]):
            break
        
        if first:
            if z != int(info[0]):
                days[x][z - 1] = " "
            else:
                days[x][z - 1] = str(current)
                current += 1
                first = False
        else:
            days[x][z - 1] = str(current)
            current += 1

print("Sun Mon Tue Wed Thr Fri Sat")
for x in range(0, len(days)):
    lendays = len(days[x])
    
    for z in range(0, lendays):
        if days[x][z] == 0:
            break
        elif z == 0:
            start = ""
        else:
            start = " "
        
        if len(str(days[x][z])) == 2:
            print(" " + start + str(days[x][z]), end = "")
        else:
            print("  " + start + str(days[x][z]), end = "")
    print("", end = "\n")

File: Contests/CCC/CCC '00 J1 - Calendar.cpp
#include <stdio.h>

int main() {
    int s, a;
    scanf("%d %d", &s, &a);
    printf("Sun Mon Tue Wed Thr Fri Sat\n");

    for (int i = 1; i < s; i++)
        printf("    ");

    for (int i = 1; i <= a; i++) {
        if (i >= 10) printf(" %d", i);
        else printf("  %d", i);

        if ((s + i - 1) % 7 == 0 || i == a)
            printf("\n");
        else printf(" ");
    }

    return 0;
}